Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -11,7 +11,10 @@ class ChatClientUpdater<ExtraData: ExtraDataTypes> {
         self.client = client
     }
 
-    func prepareEnvironment(newToken: Token) throws {
+    func prepareEnvironment(
+        userInfo: UserInfo<ExtraData>?,
+        newToken: Token
+    ) throws {
```

Comment:
shouldn't we also reconnect if `userInfo` has changed?

Reply:
A good question, in my understanding the answer is no because we used to live without passing user info at all, so we only care if the user or token have changed